### PR TITLE
samples: lwm2m_client: update overlays

### PR DESCRIPTION
**Pull request by VeijoPesonen**
* Improve the overlay documentation
* Adds low-power-overlay which disables serial port

### CODE REVIEW

**Comment by VeijoPesonen** [comment]
@umapraseeda, @jheiskan81 Please review.

**Review by rlubos — APPROVED** [review]

File: samples/nrf9160/lwm2m_client/sample_description.rst [hunk 1]
```diff
@@ -521,12 +522,15 @@ The following files are available:
...
 * :file:`overlay-assist-cell.conf` - Enables cell-based location assistance.
 * :file:`overlay-assist-pgps.conf` - Enables P-GPS assistance in the sample.
+* :file:`overlay-lowpower.conf` - Disables certain features to bring the power consumption down.
 
 Moreover, the sample also provides the following files for LwM2M 1.1 features:
 
-* :file:`overlay-lwm2m-1.1.conf`
-* :file:`overlay-lwm2m-1.1-core-interop.conf`
-* :file:`overlay-lwm2m-1.1-object-interop.conf`
+* :file:`overlay-lwm2m-1.1.conf` - Enables v1.1. Additionally makes it possible to run Interoperability Test Cases [0-499].
```
[the remark this answers is not in the document]
Comment (VeijoPesonen) [reply samples/nrf9160/lwm2m_client/sample_description.rst:529]: The overlay makes it possible to run the tests but it won't start execution.

**Comment by VeijoPesonen** [comment]
@umapraseeda I applied all your suggestions.

**Review by umapraseeda — APPROVED** [review]

**Comment by VeijoPesonen** [comment]
@umapraseeda I assume _doc-required_-label can be removed now. Same goes for the _changelog-entry-required_-label

**Comment by umapraseeda** [comment]
> @umapraseeda I assume _doc-required_-label can be removed now. Same goes for the _changelog-entry-required_-label

Doc required label is automatically added if the PR edits RSTs. That should stay and the PR can be merged if you have a techwriter approval. Removing changelog-required label as no entry is required in the changelog (discussed and agreed offline)

**Pull request merged**